src/special/commitizen.js

```diff
@@ -0,0 +1,22 @@
+import path from 'path';
+
+export default function parseCommitizen(content, filePath, deps, rootDir) {
+  const packageJsonPath = path.resolve(rootDir, 'package.json');
+  const resolvedFilePath = path.resolve(filePath);
+
+  if (resolvedFilePath === packageJsonPath) {
```

Comment:
One more thing, you should check the `commitizen` package itself is inside the `deps` array at the first beginning.